core/cache/controller.go

```diff
@@ -80,23 +88,27 @@ func (c *Controller) loop() error {
 			case ApplicationChange:
 				c.updateApplication(ch)
 			case RemoveApplication:
-				c.removeApplication(ch)
+				err = c.removeApplication(ch)
 			case CharmChange:
 				c.updateCharm(ch)
 			case RemoveCharm:
-				c.removeCharm(ch)
+				err = c.removeCharm(ch)
 			case MachineChange:
 				c.updateMachine(ch)
 			case RemoveMachine:
-				c.removeMachine(ch)
+				err = c.removeMachine(ch)
 			case UnitChange:
 				c.updateUnit(ch)
 			case RemoveUnit:
-				c.removeUnit(ch)
+				err = c.removeUnit(ch)
 			}
 			if c.config.Notify != nil {
 				c.config.Notify(change)
 			}
+
+			if err != nil {
```

Comment:
Wasn't sure what to do here. Maybe log instead of trashing the loop?
As it is an entity that has an error doing clean-up is not evicted/deregistered.

Reply:
I think just logging as an error for now is fine.